dhcpv6/dhcpv6message.go

```diff
@@ -198,26 +198,32 @@ func NewRequestFromAdvertise(advertise DHCPv6, modifiers ...Modifier) (DHCPv6, e
 	return d, nil
 }
 
-// NewReplyFromRebind creates a new REPLY packet based on a REBIND packet.
-func NewReplyFromRebind(rebind DHCPv6, modifiers ...Modifier) (DHCPv6, error) {
-	if rebind == nil {
-		return nil, errors.New("REBIND cannot be nil")
-	}
-	if rebind.Type() != REBIND {
-		return nil, errors.New("The passed REBIND must have REBIND type set")
-	}
-	reb, ok := rebind.(*DHCPv6Message)
+// NewReplyFromMessage creates a new REPLY packet based on a MESSAGE packet.
```

Comment:
s/Message/DHCPv6Message/ is that what you mean? There is no MESSAGE packet type, so that would look a bit misleading. Alternatively you can mention explicitly all the message types that are considered for creating the REPLY message (i.e. solicit, renew, etc)